### PR TITLE
fix(cli): smooth remote pull progress bar speed and ETA

### LINKED ISSUE #52: Remote pull progress bar is jerky: speed drops to 0 B/s between server updates
## Description

When pulling a model through the server (`mold pull` or auto-pull during `mold run` in remote mode), the download progress bar behaves erratically:

- Speed oscillates between real values (e.g. `36.99 MiB/s`) and `0 B/s`
- ETA jumps wildly (e.g. `2m` → `0s` → `2d` → `15m`)
- Progress appears to reset/stall between server status updates
- The bar is not smooth or normalized like a local `mold pull`

## Reproduction

```
mold run wuerstchen-v2:fp16 "a castle"
# (model not on server, triggers remote pull)
```

Output shows the progress bar jumping:
```
[1/7] ...diffusion_pytorch_model.safetensors   [╸──────] 9.26 MiB/3.93 GiB (36.99 MiB/s, 2m)
[1/7] ...diffusion_pytorch_model.safetensors   [╸──────] 7.11 MiB/3.93 GiB (28.37 MiB/s, 2m)
[1/7] ...diffusion_pytorch_model.safetensors   [╸──────] 4.23 KiB/3.93 GiB (0 B/s, 0s)
[1/7] ...diffusion_pytorch_model.safetensors   [╸──────] 1.57 MiB/3.93 GiB (4.44 MiB/s, 15m)
[1/7] ...diffusion_pytorch_model.safetensors   [╸──────] 16.00 KiB/3.93 GiB (0 B/s, 0s)
```

## Likely Cause

The CLI polls the server's `/api/models/pull` SSE stream for progress events. Between server-sent updates, the `indicatif` progress bar has no new data, so:

1. The bar's internal rate estimator sees zero bytes transferred → reports `0 B/s`
2. When the next server event arrives with a large byte jump, the instantaneous rate spikes
3. The position may appear to go backwards if the server reports cumulative bytes differently than expected

## Possible Fixes

- **Smooth the rate display**: Use `indicatif`'s built-in ETA smoothing (`ProgressStyle::with_key("smoothed_eta", ...)`) or a moving-average window for speed
- **Don't reset position between updates**: Ensure cumulative progress is only ever incremented, never set to a lower value
- **Interpolate between server events**: If the server sends updates every N seconds, the CLI could interpolate progress between events to keep the bar moving smoothly
- **Increase server-side update frequency**: Send progress events more frequently from the server's download handler (e.g. every 100ms instead of every chunk)

## Context

Local `mold pull` works smoothly because `indicatif` gets progress callbacks directly from `hf-hub`'s download stream with fine-grained byte counts. The remote path goes through an SSE intermediary which batches/delays updates.

### PR DESCRIPTION
**Pull request by jamesbrink**
## Summary

- Replace indicatif's built-in `{bytes_per_sec}` and `{eta}` with a custom `SmoothedRate` sliding-window estimator (8 samples, ~2 seconds) that averages download rate across recent SSE events
- Add monotonic position guard to prevent the bar from visually going backward on out-of-order events
- Enable `steady_tick(100ms)` on download bars so they redraw between coarse SSE updates
- Add `format_speed()` and `format_eta()` helpers for human-readable rate/ETA display

Closes #52

## Test plan

- [x] 16 new unit tests for `SmoothedRate` (empty, single sample, rate computation, ETA, eviction, non-negative) and formatting helpers (all magnitude tiers, edge cases)
- [x] `cargo clippy --workspace -- -D warnings` clean
- [x] `cargo fmt --check` clean
- [x] `cargo test --workspace` all 96+ tests pass
- [ ] Manual: `mold pull <model>` against remote server — verify smooth speed/ETA, no `0 B/s` flicker

### CODE REVIEW

File: crates/mold-cli/src/ui.rs [hunk 1]
```diff
@@ -376,4 +466,124 @@ mod tests {
...
+        for i in 0..10 {
+            rate.samples
+                .push_back((now + Duration::from_millis(i * 250), i * 100));
+        }
+        // manually evict as record() would
+        while rate.samples.len() > rate.max_samples {
+            rate.samples.pop_front();
+        }
+        assert_eq!(rate.samples.len(), 4);
+    }
```
[the remark this answers is not in the document]
Comment (jamesbrink) [reply crates/mold-cli/src/ui.rs]: Fixed — the test now drives through `record()` instead of pushing directly into the internal `VecDeque`.
